js/web/web-backend.js

```diff
@@ -11,22 +11,23 @@ import snarkyJsWebSrc from 'string:../../../web_bindings/snarky_js_web.bc.js';
 export { initSnarkyJS, withThreadPool };
 
 let wasm = plonkWasm();
+globalThis.plonk_wasm = wasm;
+
 let init = wasm.default;
 /**
- * @type {Worker}
+ * @type {Promise<Worker>}
+ */
+let workerPromise;
+/**
+ * @type {number | undefined}
  */
-let worker;
+let numWorkers = undefined;
 
 async function initSnarkyJS() {
   const memory = allocateWasmMemoryForUserAgent(navigator.userAgent);
   await init(undefined, memory);
 
   let module = init.__wbindgen_wasm_module;
-  let numWorkers = await getEfficientNumWorkers();
-
-  worker = inlineWorker(srcFromFunctionModule(mainWorker));
-  await workerCall(worker, 'start', { memory, module, numWorkers });
-  globalThis.plonk_wasm = overrideBindings(wasm, worker);
```

Comment:
change 2 to this file: creating the main worker and waiting for its promise to return (which is after all its child workers are started) is very slow. Therefore, we do this in a way that is not blocking our top-level await. This is fine, the main worker will only be needed when we do our first expensive operation and we await the promise at that time